src/Datasource/EntityTrait.php

```diff
@@ -360,6 +360,20 @@ public function unsetProperty($property)
     }
 
     /**
+     * Get the list of all properties.
+     *
+     * The list of all properties is all standard properties
+     * plus virtual properties plus hidden properties.
+     *
+     * @return array A list of all entity properties
+     */
+    public function allProperties()
```

Comment:
That's what `allProperties()` is supposed to be. Another method which returns data similar to `toArray()` but including hidden properties.